barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_scalar.test.cpp

```diff
@@ -54,7 +54,7 @@ TYPED_TEST(CycleScalarTest, TestFromWitness)
 
     EXPECT_EQ(ScalarField(reconstructed), scalar_val);
 
-    check_circuit_and_gate_count(builder, 0);
+    check_circuit_and_gate_count(builder, 2761);
```

Comment:
Big jump from now performing primality check during cycle scalar construction. Note that the first primality check has a large cost due to the range constraint mechanism. Subsequent checks are much cheaper.